include/ensmallen_bits/problems/schaffer_function_n1.hpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file schaffer_function_n1.hpp
+ * @author Sayan Goswami
+ *
+ * Implementation of Schaffer function N.1.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_PROBLEMS_SCHAFFER_FUNCTION_N1_HPP
+#define ENSMALLEN_PROBLEMS_SCHAFFER_FUNCTION_N1_HPP
+
+namespace ens {
+namespace test {
+
+/**
+ * The Schaffer function N.1 is defined by
+ *
+ * \f[
+ * f_1(x) = x^2
+ * f_2(x) = (x-2)^2
+ * \f]
+ *
+ * The optimal solutions to this multi-objective function lie in the
+ * range [0, 2].
+ *
+ * @tparam arma::mat Type of matrix to optimize.
+ */
+template<typename MatType = arma::mat>
+class SchafferFunctionN1
```

Comment:
Can you add one or two more functions we can use for further testing?

Reply:
Sure, will do!

Reply:
Marcus, I have added the Fonsec Flemming test function too.

Reply:
Nice, let me know if you need any help with the debug process.

Reply:
Yes, thanks! :)
I am adding a couple of other test functions too to see if this is a one-off error.